### PR TITLE
Improved Testing

### PR DESCRIPTION
**Pull request by rsimmons1**
Added tests for the tutorial and spa models to check that they are functioning. Cleaned up and added functionality to testing_tools.py, created testing docs and removed test_example.py from travisCI testing stack.

### CODE REVIEW

File: nengo_gui/testing_tools.py [hunk 1]
```diff
@@ -42,7 +41,7 @@ def start_stop_sim(driver):
     play_button = driver.find_element_by_xpath('//*[@id="pause_button"]')
     play_button.click()
 
-def folder_location(var_path):
+def folder_location(var_path,indiv_file=None):
```
Comment (Seanny123) [inline nengo_gui/testing_tools.py]: spacing

File: nengo_gui/testing_tools.py [hunk 2]
```diff
@@ -71,6 +70,8 @@ def folder_location(var_path):
     test_files = os.listdir(test_folder)
     test_files = filter((lambda x: ((x.count('.') == 1) and ('.py' in x))),
         test_files)
+    if(indiv_file != None):
```
Comment (Seanny123) [inline nengo_gui/testing_tools.py]: no parentheses (did you run this code through flake8?)

**Comment by Seanny123** [comment]
LGTM, except for the few style issues.

**Comment by rsimmons1** [comment]
Just went through flake8 lots of little style issues everywhere, all correct now.  I'll remember to go through flake8 really helps with these small style issues.

**Pull request merged**